src/awkward/_connect/numba/growablebuffer.py

```diff
@@ -0,0 +1,407 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import numba
+import numba.core.typing.npydecl
+import numpy
+
+
+class GrowableBuffer:
+    def __init__(self, dtype, *, initial=1024, resize=8.0):
```

Comment:
I'm not sure about the default value of `resize=8.0`. I agree, it should be large enough. Does it have to be a `Float`?

Reply:
With the algorithm, "first panel is small, all the rest are much larger but equal to each other," the factor separating the first panel size from the rest should be a fairly large number. I was debating between 8 and 16; they're both an order of magnitude, but 8 is `2**3` and 3 is an odd number, whereas 4 is evenly half of a byte. (Shrug.)

This would ideally be set by a performance study. But that performance study would have to use some realistic test cases, since the thing that the new resize algorithm is trying to model is the steep distribution of "many small buffers, a few large ones." The actual distribution of number of small buffers (how small?) to number of large buffers (how large?) would matter.

Now that it's large, you're right that it no longer needs to be a floating point number. In the old algorithm, there's some literature saying that

$$\phi = \frac{1 + \sqrt{5}}{2} \approx 1.5$$

is the right `resize` factor, so that's why it was a float. We want both the C++ GrowableBuffer and the Numba GrowableBuffer to use the same algorithm and parameters, and if the C++ algorithm still needs to be changed (I don't know whether it does or not), leaving it as a floating point number but changing the algorithm would be the easiest change. It appears in a lot of places, just to be passed from one function to another, so changing the data type is hard, but including the `if (panels_.size() == 1)` clause in one place is easy.

Reply:
Thanks, @jpivarski !